high_card.rb

```diff
@@ -1,5 +1,52 @@
 # Use these two arrays to generate a deck of cards.
+
+def card_game
 ranks = ["A", 2, 3, 4, 5, 6, 7, 8, 9, 10, "J", "Q", "K" ]
 suits = [ "hearts", "spades", "clubs", "diamonds" ]
 
+
+# 1. Create deck of cards:
+  card_deck = []
+  ranks.each do |r|
+    suits.each do |s|
+      card_deck.push({
+        rank: r,
+        suit: s
+      })
+    end
+  end
+
+  return card_deck
+
+# 2. Populate players:
 players = []
+
+loop do
+  puts'We have #{players.length} so far. Enter the next player name or enter \'play\' to begin.'
+  name = gets.chomp.to_s
+
+  break if name == 'play'
+  players.push(name)
+end
+
+# 3. Deal cards:
+  return card_deck.shuffle
```

Comment:
i don't think you need the return key word here, should execute the method regardless